src/app/component/eco-news/components/create-news/create-news.component.ts

```diff
@@ -118,33 +118,31 @@ export class CreateNewsComponent implements OnInit {
     if ( !filter.isActive ) {
       filter.isActive = true;
       this.isArrayEmpty = false;
-      this.createNewsForm.value.tags.push(filter.name.toLowerCase());
+      this.createNewsForm.value.tags = [...this.createNewsForm.value.tags, filter.name.toLowerCase()];
```

Comment:
Explain these changes.

Reply:
push method of the Array is mutable, it changes our array.
instead I create new Array and use spread operator to copy all items from our previous array, and add new item to end of array.
it calls Immurability.